ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/LmdGhost.java

```diff
@@ -64,41 +70,58 @@ public static BeaconBlock lmd_ghost(
         return head;
       }
 
-      // TODO: this is missing hash_tree_root(child_block).
+      UnsignedLong max_value = UnsignedLong.ZERO;
+      for (BeaconBlock child : children) {
+        UnsignedLong curr_value = get_vote_count(start_state, store, child, attestation_targets);
+        if (curr_value.compareTo(max_value) > 0) {
+          max_value = curr_value;
+        }
+      }
+
       head =
           children.stream()
+              .filter(
+                  child ->
+                      get_vote_count(start_state, store, child, attestation_targets)
+                              .compareTo(max_value)
+                          == 0)
               .max(
                   Comparator.comparing(
-                      child_block ->
-                          Math.toIntExact(
-                              get_vote_count(store, child_block, attestation_targets).longValue())))
-              .get();
+                      child -> hash_tree_root(child.toBytes()).toLong(ByteOrder.LITTLE_ENDIAN)));
```

Comment:
I'm not sure if this fulfills this comment from the spec? "Ties broken by favoring block with lexicographically higher root" Can we get another set of eyes on this? Seems to me as if ties are broken by vote count rather than higher root here, although I may be wrong.

Reply:
To clarify, the point of the code in lmd_ghost is to get the maximum looking at the get_vote_count and then if there is a tie, to break the tie with hash_tree_root